src/compiler/nir/nir_divergence_analysis.c

```diff
@@ -201,6 +201,7 @@ visit_intrinsic(bool *divergent, nir_intrinsic_instr *instr, unsigned subgroup_s
    case nir_intrinsic_shared_atomic_xor:
    case nir_intrinsic_shared_atomic_exchange:
    case nir_intrinsic_shared_atomic_comp_swap:
+   case nir_intrinsic_shuffle:
```

Comment:
is it? a shuffle of uniform should stay uniform ...

Reply:
yeah, I think there a similar issue with some other subgroup intrinsics
a uniform nir_intrinsic_shuffle could probably also be optimized into a copy

Reply:
This would need to have the NIR optimizer aware of divergence, might be an idea...

Reply:
Would be something done outside of this patch series?

Currently I've just got ACO to select some reductions into p_as_uniform/s_mov_b32/etc.